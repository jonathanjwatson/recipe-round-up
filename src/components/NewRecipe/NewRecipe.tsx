import React, { useEffect, useState } from "react";
import API from "../../API/API";
import ErrorBoundary from "../ErrorBoundary/ErrorBoundary";

const NewRecipe: React.FC = () => {
  const [recipes, setRecipes] = useState([
    {
      id: null,
      name: "",
      items: []
    }
  ]);

  useEffect(() => {
    API.getRecipes()
      .then(response => {
        //TODO: Create story for back-end team to fix typo: "quntity".
        setRecipes(response.data.recipes);
      })
      .catch(err => {
        console.log(err);
      });
  }, []);

  return (
    <>
      <h1>New Recipe</h1>
      <ErrorBoundary>
        <table className="table">
          <thead className="thead-dark">
            <tr>
              <th scope="col">Recipe Id</th>
              <th scope="col">Recipe Name</th>
              <th scope="col">Required Ingredients</th>
              <th scope="col"></th>
            </tr>
          </thead>
          <tbody>
            {recipes.map((recipe, i: number) => (
              <tr key={i}>
                <th scope="row">{recipe.id}</th>
                <th scope="row">{recipe.name}</th>
                <th scope="row">
                  {recipe.id && <span>Loop through items</span>}
                </th>
                <th scope="row">
                  {recipe.id && (
                    <button className="btn btn-dark">Create Order</button>
                  )}
                </th>
              </tr>
            ))}
          </tbody>
        </table>
      </ErrorBoundary>
    </>
  );
};

export default NewRecipe;
