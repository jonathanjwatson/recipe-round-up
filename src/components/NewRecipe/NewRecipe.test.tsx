import React from "react";
import { render, cleanup } from '@testing-library/react';
import NewRecipe from "./NewRecipe";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<NewRecipe />);
  expect(asFragment()).toMatchSnapshot();
});
