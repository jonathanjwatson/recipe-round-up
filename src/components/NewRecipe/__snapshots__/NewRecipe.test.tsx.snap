// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <h1>
    New Recipe
  </h1>
  <table
    class="table"
  >
    <thead
      class="thead-dark"
    >
      <tr>
        <th
          scope="col"
        >
          Recipe Id
        </th>
        <th
          scope="col"
        >
          Recipe Name
        </th>
        <th
          scope="col"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        />
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
